### ISSUE TITLE
Nondeterministic train test split

### ISSUE THREAD

**Issue by trangdata** [EpistasisLab/tpot#933]
The default `random_state = None` in `TPOTClassifier()` or `TPOTRegressor()` may result in over-optimistic accuracy of TPOT-recommended pipeline.

## Context
The first example [here](https://github.com/EpistasisLab/tpot/blob/815b0e28c7c8c4935363298bb6ce4713d280cece/docs_sources/examples.md#iris-flower-classification) does not specify a value for the `random_state` argument in either `train_test_split()` or `TPOTClassifier()`. Consequently, in exported file `tpot_iris_pipeline.py`, `train_test_split()` does not have a specified seed. This likely means the two splits are different, resulting in the TPOT run essentially having access to part of the test set. In other words, the hyperparameters were obtained knowing some of the test data.

This is not an issue when training and testing set are read in separately, and the exported file is modified accordingly.

## Possible fix

Should we require the same seed for `train_test_split` and `TPOTClassifier()` or `TPOTRegressor()`?

Or perhaps let the user read ini the test and train set separately?

**Comment by weixuanfu** [comment]
The issue has been fixed in a new version of TPOT (v0.11.0). Please feel free to reopen it if you have any questions or suggestions.
